XenAdmin/Dialogs/ManageCPUGroupDialog.Designer.cs

```diff
@@ -0,0 +1,300 @@
+            this.panel1.Padding = new System.Windows.Forms.Padding(1);
+            this.panel1.Size = new System.Drawing.Size(721, 30);
+            this.panel1.TabIndex = 67;
+            // 
+            // toolStrip1
+            // 
+            this.toolStrip1.AutoSize = false;
+            this.toolStrip1.BackColor = System.Drawing.SystemColors.InactiveBorder;
+            this.toolStrip1.Dock = System.Windows.Forms.DockStyle.Fill;
+            this.toolStrip1.Font = new System.Drawing.Font("Segoe UI", 9F);
+            this.toolStrip1.GripStyle = System.Windows.Forms.ToolStripGripStyle.Hidden;
+            this.toolStrip1.Items.AddRange(new System.Windows.Forms.ToolStripItem[] {
+            this.toolStripButtonNew,
+            this.toolStripButtonDelete,
+            this.toolStripButtonEddit});
+            this.toolStrip1.Location = new System.Drawing.Point(1, 1);
+            this.toolStrip1.Name = "toolStrip1";
+            this.toolStrip1.Size = new System.Drawing.Size(719, 28);
+            this.toolStrip1.TabIndex = 65;
+            this.toolStrip1.Text = "toolStrip1";
+            // 
+            // toolStripButtonNew
+            // 
+            this.toolStripButtonNew.Font = new System.Drawing.Font("Segoe UI", 9F);
+            this.toolStripButtonNew.Image = global::XenAdmin.Properties.Resources._000_NewVirtualAppliance_h32bit_16;
+            this.toolStripButtonNew.ImageTransparentColor = System.Drawing.Color.Magenta;
+            this.toolStripButtonNew.Name = "toolStripButtonNew";
+            this.toolStripButtonNew.Size = new System.Drawing.Size(122, 25);
+            this.toolStripButtonNew.Text = "&New CPU Group...";
+            this.toolStripButtonNew.Click += new System.EventHandler(this.toolStripButtonNew_Click);
+            // 
+            // toolStripButtonDelete
+            // 
+            this.toolStripButtonDelete.Font = new System.Drawing.Font("Segoe UI", 9F);
+            this.toolStripButtonDelete.Image = global::XenAdmin.Properties.Resources._000_DeleteVirtualAppliance_h32bit_16;
+            this.toolStripButtonDelete.ImageTransparentColor = System.Drawing.Color.Magenta;
+            this.toolStripButtonDelete.Name = "toolStripButtonDelete";
+            this.toolStripButtonDelete.Size = new System.Drawing.Size(60, 25);
+            this.toolStripButtonDelete.Text = "&Delete";
+            // 
+            // toolStripButtonEddit
+            // 
+            this.toolStripButtonEddit.Font = new System.Drawing.Font("Segoe UI", 9F);
+            this.toolStripButtonEddit.Image = global::XenAdmin.Properties.Resources.edit_16;
+            this.toolStripButtonEddit.ImageTransparentColor = System.Drawing.Color.Magenta;
+            this.toolStripButtonEddit.Name = "toolStripButtonEddit";
+            this.toolStripButtonEddit.Size = new System.Drawing.Size(80, 25);
+            this.toolStripButtonEddit.Text = "&Properties";
+            // 
+            // tableLayoutPanel1

```

Comment:
I think tableLayoutPanel1 has 1 row too many, the top row does not seem to be used.
General tip: having extra rows/columns happens very often when they are set to autosize. It may not be a problem in the first instance, but it may make it tricky to add more controls in the panel in the future. One way to avoid this is to set originally all the columns/rows to Percent, add the controls you want, and then set the necessary column/rows to Autosize.

Reply:
Thanks, I am fresh to XenCenter, so there may be some mistakes. Thank you very much.